src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -102,6 +103,76 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
             DecorateOperationSets();
         }
 
+        // Initialize ReourceData, Models and resource manager common types
+        private void InitializeModels()
+        {
+            _models = new Dictionary<Schema, TypeProvider>();
+            _resourceModels = new Dictionary<Schema, TypeProvider>();
+
+            // first, construct models and resource data models
+            foreach (var schema in _allSchemas)
+            {
+                if (_resourceDataSchemaNameToOperationSets.ContainsKey(schema.Name))
+                {
+                    var model = BuildResourceModel(schema);
+                    _resourceModels.Add(schema, model);
+                    _nameToTypeProvider.Add(schema.Name, model); // TODO: ADO #5829 create new dictionary that allows look-up with multiple key types to eliminate duplicate dictionaries
```

Comment:
Wouldn't the keys `schema` and `schema.Name` be semantically the same?

Reply:
I think so. I moved the codes from other places. `_resourceModels` and `_nameToTypeProvider` are used in different scenarios in which one used `schema` as key and another use `schema.Name` as key. It would be better we refactor them in a separate PR.